### ISSUE TITLE
Set username to a socket (client) connection

### ISSUE THREAD

**Issue by kdzekov** [ratchetphp/Ratchet#284]
Hello,

I wonder whether it is possible to attach or set username to a socket connection / connectionInterface when you make a socket connection through the client interface?
If you can, how can you get that username from the connectionInterface on the server side? 

(something like: $client->resourceId, where $client instanceof ConnectionInterface but with an username) 

Thank you!

**Comment by kinncj** [comment]
If it's using the `AbstractConnectionDecorator`, it implements `__set` and `__get`, although I can't see as a good practice since you are relying in the `ConnectionInterface` and the received instance may not implement the `__set` and `__get` property accessors...
eg: `WampConnection` extends from `AbstractConnectionDecorator`, `IoConnection` implements `ConnectionInterface`.

In theory, your application is the one responsible to have known about the user, trough `session`, `headers`, or even `JWT`.

If you really need to *authenticate/authorize/identify* a `User`, I'd rater do that trough a message then connection.

1. User opens a connection
2. User may send a message with its credentials
  1. Save the user state (eg: save authenticated in the pool, or anywhere)
  2. If user is authenticated/authorized/identified
    1. User can send messages to other channels

Basically, you will be implementing *authentication/authorization/identification* trough sockets, which sounds awesome.

**Comment by kdzekov** [comment]
Hi,

well, i did it using onMessage event. So i have something like:

socket.onconnect = function(e)
{
...
socket.send(username)
}

it works well. I had similar idea before but i thought maybe if there is more direct way.
Thanks for the help!

**Comment by kinncj** [comment]
:+1:

**Comment by harveyslash** [comment]
what happens if client connects and doesn't send any message? the connection will be established forever?

**Comment by cboden** [comment]
@harveyslash yes
